strapi/strapi
# fix #11845 contains and limit bug for relationship field

**Note:** This PR isn't tested.

See #11845 for more info.

### What does it do?

changes

```js
const params = { _limit: state._limit, ...defaultParams };

if (state._contains) {
  params[containsKey] = state._contains;
}
```

to

```js
const params = { limit: state._limit, ...defaultParams };

if (state._contains) {
  const [fieldName] = containsKey.split('_')
  params[`filters[${fieldName}][$contains]`] = state._contains;
}
```

### Why is it needed?

Page crashes when fetching large amount of related records.

### How to test it?

1. Create a relationship field
2. Create more than 20 records in the relationship table
3. Go to the record in which you can link records from the relationship table.
4. Inspect the request being done.
5. See that more than 20 records are being returned despite the _limit param being passed to the request.

### Related issue(s)/PR(s)

#11845
